### PR TITLE
fix(account-lib): fix addSignature method of cspr transaction

### PR DESCRIPTION
**Pull request by CapnMigraine**
Set signature on transaction deploy

Ticket: STLX-2076

### CODE REVIEW

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 1]
```diff
@@ -36,14 +37,28 @@ export class Transaction extends BaseTransaction {
...
+   * @param {KeyPair} keyPair The key pair that created the signature
    */
-  addSignature(signature: string): void {
+  addSignature(signature: string, keyPair: KeyPair): void {
+    const pub = keyPair.getKeys().pub;
+    const signatureBuffer = Uint8Array.from(Buffer.from(signature, 'hex'));
+    const pubKeyBuffer = Uint8Array.from(Buffer.from(pub, 'hex'));
+    const parsedPublicKey = Keys.Secp256K1.parsePublicKey(pubKeyBuffer, 'raw');
+    const pubKeyHex = Keys.Secp256K1.accountHex(parsedPublicKey);
+    if (pubKeyHex.slice(2) !== _.toLower(pub)) {
```
Comment (ipelaez-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:54]: Hello Luis, i dont remeber which prefix does remove here, it is secpt256k1 doesnt it? could you clarify this?
I think you can use SECP256K1_PREFIX from [account-lib/src/coin/cspr/constants.ts)](https://github.com/BitGo/BitGoJS/blob/STLX-2076-cspr-fix-addSignature/modules/account-lib/src/coin/cspr/constants.ts)

File: modules/account-lib/src/coin/cspr/transaction.ts [hunk 2]
```diff
@@ -36,14 +37,28 @@ export class Transaction extends BaseTransaction {
...
+      throw new SigningError('Signer does not match signature');
+    }
+    const signedDeploy = DeployUtil.setSignature(this._deploy, signatureBuffer, PublicKey.fromSecp256K1(parsedPublicKey))
+    const approval = _.last(signedDeploy.approvals) as Approval;
+    if (approval.signature.slice(2) !== signature) {
```
Comment (ipelaez-simtlix) [inline modules/account-lib/src/coin/cspr/transaction.ts:59]: Same as above

File: modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts [hunk 3]
```diff
@@ -85,4 +86,38 @@ describe('Casper Transaction Builder', () => {
...
+      const sig = '0072f40621663fd03c5e13b413d5c354cdf4c7e76672aa61fd8ede0f1ac09f5de107d725eb40e1efb9037940d74ef9b2efaa1d66d0991a5322639481c2d4280775';
+      const pub = '03dca7d5d68fba12a604e992a47504d10e6795cdc6db438abb741788c71c4b7428';
+      const signingKeyPair = new KeyPair({ pub });
+      builder.signature(sig, signingKeyPair);
+      const tx = (await builder.build()) as Transaction;
+      const signer = tx.casperTx.approvals[0].signer;
+      const signature = tx.casperTx.approvals[0].signature;
+
+      signer.slice(2).should.equal(pub);
+      signature.slice(2).should.equal(_.toLower(sig));
```
Comment (ipelaez-simtlix) [inline modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts:115]: Same as above,

**Review by ipelaez-simtlix — COMMENTED** [review]
Seems good overall, just a small change

File: modules/account-lib/src/coin/cspr/transaction.ts (hunk 1 above)
Comment (CapnMigraine) [reply modules/account-lib/src/coin/cspr/transaction.ts]: Added a new helper function in utils to centralize this logic

File: modules/account-lib/src/coin/cspr/transaction.ts (hunk 2 above)
Comment (CapnMigraine) [reply modules/account-lib/src/coin/cspr/transaction.ts]: done

File: modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts (hunk 3 above)
Comment (CapnMigraine) [reply modules/account-lib/test/unit/coin/cspr/transactionBuilder/transactionBuilder.ts]: done

**Comment by ipelaez-simtlix** [comment]
LGTM

**Review by ipelaez-simtlix — APPROVED** [review]
LGTM

**Pull request merged**